rust/lance-io/src/lib.rs

```diff
@@ -200,6 +200,18 @@ impl ReadBatchParams {
             )),
         }
     }
+
+    pub fn to_offsets_total(&self, total: u32) -> Result<PrimitiveArray<UInt32Type>> {
```

Comment:
Why return result? I don't see any error and you just `unwrap()` it later anyways.